backend/src/main/java/ch/sectioninformatique/template/auth/AuthClient.java

```diff
@@ -0,0 +1,78 @@
+package ch.sectioninformatique.template.auth;
+
+import org.springframework.stereotype.Service;
+import org.springframework.web.reactive.function.client.WebClient;
+
+import jakarta.validation.Valid;
+import reactor.core.publisher.Mono;
+import org.springframework.http.MediaType;
+
+/**
+ * Client service for authentication operations.
+ * This service provides methods to interact with the spring-auth application's endpoints,
+ * including login and registration functionalities.
+ * It uses WebClient to perform HTTP requests and handle responses reactively.
+ */
+@Service
+public class AuthClient {
+
+    /** WebClient instance for making HTTP requests */
+    private final WebClient webClient;
+
+    /** Constructor to initialize the WebClient */
+    public AuthClient(WebClient webClient) {
+        /**
+         * TODO : We should externalize the URL in a configuration file or environment variable
+         * Attention : don't use https://auth.sectioninformatique.ch beacause the SSL certificate is not valid for this domain
+         * Use https://spring-auth.jcloud.ik-server.com/ as it is the Jelastic server domain name, with a valid SSL certificate
+        */
+        this.webClient = WebClient.create("https://spring-auth.jcloud.ik-server.com/");
+    }
+
+    /** 
+     * Performs user login by sending credentials to the authentication endpoint.
+     * 
+     * @param credentialsDto The CredentialsDto containing user login data
+     * @return A Mono<String> containing the authentication response (e.g., token or status message)
+     */
+    public Mono<String> login(@Valid CredentialsDto credentialsDto) {
```

Comment:
Why should we use a String return value ? Would it be possible to return the complete ResponseEntity sent by the Spring-Auth endpoint ?